### PR TITLE
fix: set correct error reason for non-existing gen account

### PR DESCRIPTION
**Pull request by AnExsomnis**

### CODE REVIEW

File: core/execution/market.go [hunk 1]
```diff
@@ -1267,7 +1267,7 @@ func (m *Market) validateAccounts(ctx context.Context, order *types.Order) error
 	if !m.collateral.HasGeneralAccount(order.Party, asset) {
 		// adding order to the buffer first
 		order.Status = types.OrderStatusRejected
-		order.Reason = types.OrderErrorInsufficientAssetBalance
+		order.Reason = types.OrderErrorMissingGeneralAccount
```
Comment (wwestgarth) [inline core/execution/market.go:1270]: It looks like the spec was updated such that we don't need to do anything for this ticket as `0024-OSTA-028` was removed ?
https://github.com/vegaprotocol/vega/issues/6951#issuecomment-1364033964
https://github.com/vegaprotocol/specs/pull/1487

The comment against `ORDER_ERROR_INSUFFICIENT_ASSET_BALANCE` and the AC `0024-OSTA-027` mentions "having no general account"
```
// Party has an insufficient balance, or does not have
// a general account to submit the order (no deposits made
// for the required asset)
ORDER_ERROR_INSUFFICIENT_ASSET_BALANCE = 43;
```

Comment (EVODelavega) [reply core/execution/market.go:1270]: Hmm, that makes sense. An error saying _"the general account is missing"_ doesn't really make sense given that a party can have N general accounts (1 per asset). Functionally, there's no difference between a non-existent general balance for a given asset, and a general account with balance 0.

Comment (AnExsomnis) [reply core/execution/market.go:1270]: @wwestgarth  thanks, i didnt see the commen, closed the issue now

Comment (AnExsomnis) [reply core/execution/market.go:1270]: Not sure the ORDER_ERROR_INSUFFICIENT_ASSET_BALANCE is the best way of saying that no general account exists, just after checking if there was:
``` if !m.collateral.HasGeneralAccount(order.Party, asset) ... ```
Am I interpreting this wrong? Please let me know

Comment (EVODelavega) [reply core/execution/market.go:1270]: I don't think a different error message is going to help things long term. True, it makes sense in the conventional account/balance modal, but from a UX perspective, at least for the foreseeable future, parties will be transferring tokens from their crypto wallets. In that context, being told that _"you don't have a wallet for token X"_ means something very different to _"wallet has balance 0 for token X"_.

To address such an issue, the only thing a party has to do is transfer some balance, whether the account existed or its balance was zero makes no difference. To make a distinction between having no balance, or never having had any balance for a certain asset seems rather arbitrary, and IMO can only really cause confusion

**Pull request closed without merging**